### PR TITLE
Super dHEDGE Factory Version

### PR DESCRIPTION
**Pull request by rashtrakoff**
This PR has been created specifically for Superfluid review process. This **should not** be merged to `master` branch.

### CODE REVIEW

File: contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol [hunk 1]
```diff
@@ -158,6 +158,7 @@ library dHedgeHelper {
             (, int96 _flowRate) = tokenData.superToken.getFlow(_sender);
             uint256 _currFlowRate = uint256(uint96(_flowRate));
             
+            // Not really necessary ?
             assert(
```
Comment (ngmachado) [inline contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol]: I think is a good practice to check invariances. I would leave it.

Comment (ngmachado) [reply contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol]: Check the if else block : _depositAmount > _userUninvested. What should happen when _depositAmount = _userUninvested?

File: contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+            _depositBalance -= _feeCollected;
+
+            // Perform deposit transaction iff amount of underlying tokens is greater than 0
+            if (_depositBalance > 0) {
+                // Store the timestamp of last time a deposit & distribution was made
+                tokenData.lastDepositAt = block.timestamp;
+
+                // Transfer the fees collected to the owner
+                IERC20(_depositToken).safeTransfer(
```
Comment (ngmachado) [inline contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol]: if _feeCollected is zero don't transfer

Comment (rashtrakoff) [reply contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol]: Yes, changes have been made.

File: contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol (hunk 1 above)
Comment (rashtrakoff) [reply contracts/dHedge-Factory-Version/Libraries/dHedgeHelper.sol]: Ideally nothing should happen as _depositAmount == _userUninvested is only possible if a user updates a stream with the same stream rate as before updation. For example if my current stream rate is $10/hr and I send a transaction to update this to $10/hr then nothing should happen.

**Pull request closed without merging**